homeassistant/components/pcal9535a/switch.py

```diff
@@ -0,0 +1,103 @@
+"""Support for switch sensor using I2C PCAL9535A chip."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.const import DEVICE_DEFAULT_NAME
+from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_INVERT_LOGIC = "invert_logic"
+CONF_I2C_ADDRESS = "i2c_address"
+CONF_I2C_BUS = "i2c_bus"
+CONF_PINS = "pins"
+CONF_STRENGTH = "strength"
+
+STRENGTH_025 = "0.25"
+STRENGTH_050 = "0.5"
+STRENGTH_075 = "0.75"
+STRENGTH_100 = "1.0"
+
+DEFAULT_INVERT_LOGIC = False
+DEFAULT_I2C_ADDRESS = 0x20
+DEFAULT_I2C_BUS = 1
+DEFAULT_STRENGTH = STRENGTH_100
+
+_SWITCHES_SCHEMA = vol.Schema({cv.positive_int: cv.string})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_PINS): _SWITCHES_SCHEMA,
+        vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
+        vol.Optional(CONF_STRENGTH, default=DEFAULT_STRENGTH): vol.In(
+            [STRENGTH_025, STRENGTH_050, STRENGTH_075, STRENGTH_100]
+        ),
+        vol.Optional(CONF_I2C_ADDRESS, default=DEFAULT_I2C_ADDRESS): vol.Coerce(int),
+        vol.Optional(CONF_I2C_BUS, default=DEFAULT_I2C_BUS): cv.positive_int,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the PCAL9535A devices."""
+    from pcal9535a import PCAL9535A
+
+    invert_logic = config.get(CONF_INVERT_LOGIC)
+    i2c_address = config.get(CONF_I2C_ADDRESS)
+    bus = config.get(CONF_I2C_BUS)
+
+    pcal = PCAL9535A(bus, i2c_address)
+
+    switches = []
+    pins = config.get(CONF_PINS)
+    for pin_num, pin_name in pins.items():
+        pin = pcal.get_pin(pin_num // 8, pin_num % 8)
+        switches.append(PCAL9535ASwitch(pin_name, pin, invert_logic))
+    add_entities(switches)
+
+
+class PCAL9535ASwitch(ToggleEntity):
```

Comment:
Please inherit from `SwitchDevice` imported from the switch integration instead.